src/Sparrow/Utils/TaskExecutor.cs

```diff
@@ -34,8 +36,24 @@ private void Run()
                         }
                         catch { }
                     }
-                    _event.WaitOne();
-                    _event.Reset();
+
+                    // PERF: Entering a kernel lock even if the ManualResetEventSlim will try to avoid that doing some spin locking
+                    //       is very costly. This is a hack that is allowing amortize a bit very high frequency events. The proper
+                    //       way to handle requires infrastructure changes. http://issues.hibernatingrhinos.com/issue/RavenDB-8126
+                    if (tries < 5)
+                    {
+                        // Yield execution quantum. If we are in a high-frequency event we will be able to avoid the kernel lock. 
+                        Thread.Sleep(0);
```

Comment:
IsSet shouldn't because that only reads a variable. However that still won't solve the issue. Because wait will bail anyways when someone already marked it.